### DISCUSSION TITLE
reduced move functionality in V0.74

### DISCUSSION THREAD

**Discussion by Alienheartbeat** [zim-desktop-wiki/zim-desktop-wiki#1694] [General]
Up to v0.69 I could move pages by dragging and dropping using the left side pane.
Then in 0.70 this no longer worked,
but I could still drag and drop pages by getting a page into the pathbar,
then dragging from there to the sidebar.
(I think this is issue
#github.com/zim-desktop-wiki/zim-desktop-wiki/issues/390
 )

However as of v74, this no longer works either.

So now the only thing that works is using the Move command, which requires
```	
go to the location I want to move the page to
right click, 
"Copy Location"
go to the page I want moved
right click
move
select Location field
paste in the location field
ok	
```

Which is now 9 steps to do something that used to take 2 steps (drag-drop).
I could reduce steps by typing the new location into "Location" field,
but that is real pain.

Is there any other way to do this in fewer steps that I have not discovered?

Should I raise a new issue,
or add a comment to the existing 390?

thanks
mc

**Comment by jaap-karssenberg** [comment]
Moving by drag and drop is now controlled by a preference in the page index
plugin.

Hope this helps,

Jaap

Op wo 20 okt. 2021 00:09 schreef Alienheartbeat ***@***.***>:

**Comment by Alienheartbeat** [comment]
Jaap

Thanks very much.

1.  Apologies that I did not see this.
I still cannot move pages by dragging and dropping using the left side pane,
however as in 0.73 I can drag and drop pages by getting a page into the 
pathbar,
  then dragging from there to the sidebar.

2.  However a possible minor bug:
when I moved a page, no problem,
but then when I moved it back to its original place, it appeared to be 
emptied, including headers.
Even if I moved it to several other locations first, then moved it back 
to its original location,
it appeared to be empty when returned to its original location.

However,
edit source showed the contents still there,
and a restart of Zim restored the contents.

Not serious, just fyi, as this is just something one might do when testing,

best regards and thanks for the great Zim
mc

**Comment by jaap-karssenberg** [comment]
Item 1 would be helpful to have debug output. Please file an issue for this
if you can include that info.

Item 2 is fixed in master branch, issue ticket already exists

Op wo 20 okt. 2021 07:58 schreef Alienheartbeat ***@***.***>:

**Comment by pfalcon** [comment]
> Moving by drag and drop is now controlled by a preference in the page index
plugin.

Note that in the original report, https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/1443, I didn't really argue for such a solution (enable or disable drag-n-drop moving), but instead more for an option to have an extra confirmation dialog after the drag-n-drop move (and that's what I implemented in my fork, though probably missed to submit a PR for it). Ideally, such a dialog would be on by default, but with an option to disable it for users who're sure their mouse/touchpad is never drunk and does only what they want.

**Comment by jaap-karssenberg** [comment]
I did it like this because the pattern of drag an drop followed by a dialog
feels weird. Can't think of examples in other applications. Als there is
already a dialog for moving pages that can be easily activated. But maybe a
bit more rework is needed here .

Op do 21 okt. 2021 13:33 schreef Paul Sokolovsky ***@***.***>:

**Comment by aloboa** [comment]
I would have this option in the general Preferences. Few zim users actually know the index is under the control of a plugin. And for sure I would have it "on" by default.

**Comment by Alienheartbeat** [comment]
Thanks Jaap,

I created an issue #1699
and included a clean debug log.
(clean meaning opened zim, only 1 wiki open, just did  the 2 different 
drag & drops).

Let me know if you need any further information.

regards
mc

**Comment by pfalcon** [comment]
@jaap-karssenberg:

> I did it like this because the pattern of drag an drop followed by a dialog
feels weird. Can't think of examples in other applications.

Ok, in [Claws-mail](https://www.claws-mail.org/) it's the same problem - one can inadvertently move one mail subfolder inside another (or just elsewhere), and so it asks confirmation after such a move. It doesn't feel weird for me - while drag-and-drop move operation is intuitive by itself, actual need to restructure folder tree happens rather rarely. So, in most cases when I see that confirmation dialog, I actually didn't intend to move folders, so may only appreciate the dialog saving me from the trouble to undo the move manually.

**Reply by stuaxo** [reply]
This sounds good, its safe and discoverable.
